lib/thrift/include/thrift/transport/TSSLSocket.h

```diff
@@ -0,0 +1,428 @@
+   */
+  virtual std::shared_ptr<TSSLSocket> createSocket();
+  /**
+   * Create an instance of TSSLSocket with a fresh new socket, which is interruptable.
+   */
+  virtual std::shared_ptr<TSSLSocket> createSocket(
+      std::shared_ptr<THRIFT_SOCKET> interruptListener);
+  /**
+   * Create an instance of TSSLSocket with the given socket.
+   *
+   * @param socket An existing socket.
+   */
+  virtual std::shared_ptr<TSSLSocket> createSocket(THRIFT_SOCKET socket);
+  /**
+   * Create an instance of TSSLSocket with the given socket which is interruptable.
+   *
+   * @param socket An existing socket.
+   */
+  virtual std::shared_ptr<TSSLSocket> createSocket(
+      THRIFT_SOCKET socket,
+      std::shared_ptr<THRIFT_SOCKET> interruptListener);
+  /**
+   * Create an instance of TSSLSocket.
+   *
+   * @param host  Remote host to be connected to
+   * @param port  Remote port to be connected to
+   */
+  virtual std::shared_ptr<TSSLSocket> createSocket(const std::string& host, int port);
+  /**
+   * Create an instance of TSSLSocket.
+   *
+   * @param host  Remote host to be connected to
+   * @param port  Remote port to be connected to
+   */
+  virtual std::shared_ptr<TSSLSocket> createSocket(
+      const std::string& host,
+      int port,
+      std::shared_ptr<THRIFT_SOCKET> interruptListener);
+  /**
+   * Set ciphers to be used in SSL handshake process.
+   *
+   * @param ciphers  A list of ciphers
+   */
+  virtual void ciphers(const std::string& enable);
+  /**
+   * Enable/Disable authentication.
+   *
+   * @param required Require peer to present valid certificate if true
+   */
+  virtual void authenticate(bool required);
+  /**
+   * Load server certificate.
+   *
+   * @param path   Path to the certificate file
+   * @param format Certificate file format
+   */
+  virtual void loadCertificate(const char* path, const char* format = "PEM");
+  virtual void loadCertificateFromBuffer(const char* aCertificate, const char* format = "PEM");
+  /**
+   * Load private key.
+   *
+   * @param path   Path to the private key file
+   * @param format Private key file format
+   */
+  virtual void loadPrivateKey(const char* path, const char* format = "PEM");
+  virtual void loadPrivateKeyFromBuffer(const char* aPrivateKey, const char* format = "PEM");
+  /**
+   * Load trusted certificates from specified file.
+   *
+   * @param path Path to trusted certificate file
+   */
+  virtual void loadTrustedCertificates(const char* path, const char* capath = nullptr);
+  virtual void loadTrustedCertificatesFromBuffer(const char* aCertificate,
+                                                 const char* aChain = nullptr);
+  /**
+   * Default randomize method.
+   */
+  virtual void randomize();
+  /**
+   * Override default OpenSSL password callback with getPassword().
+   */
+  void overrideDefaultPasswordCallback();
+  /**
+   * Set/Unset server mode.
+   *
+   * @param flag  Server mode if true
+   */
+  virtual void server(bool flag) {
```

Comment:
This one I would change from an inline definition to out-of-line (in the `.cpp` file).